### PR TITLE
Docs: Add block patterns list component readme

### PR DESCRIPTION
**Pull request by JustinyAhin**
## Description
Added documentation for the block patterns list component (see #22891)

## How has this been tested?
N/A, documentation only

## Types of changes
Documentation

## Checklist:
- [n/a] My code is tested.
- [n/a] My code follows the WordPress code style. 
- [n/a] My code follows the accessibility standards. 
- [n/a] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-editor/src/components/block-patterns-list/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+### Usage
+
+Renders a block patterns list.
+
+```jsx
+import { BlockPatternList } from '@wordpress/block-editor';
+
+const MyBreadcrumb = () => (
+	<BlockPatternList
+		onClickPattern={ onClick }
```
Comment (JustinyAhin) [inline packages/block-editor/src/components/block-patterns-list/README.md:25]: @ntsekouras I was wondering if it is necessary to add the `onClickPattern={ onClick }` to this code sample.

Comment (JustinyAhin) [reply packages/block-editor/src/components/block-patterns-list/README.md:25]: Also, I was wondering if there could be ways I can make these code samples more detailed. Til now, for the packages I documented, I basically added a const with a simple use case of the component.

I know these packages component are not intended to be uses by third parties, but was just wondering.

Comment (youknowriad) [reply packages/block-editor/src/components/block-patterns-list/README.md:25]: This component requires three props 

```
shownPatterns={ shownBlockPatterns }
blockPatterns={ shownBlockPatterns }
onClickPattern={ onSelectBlockPattern }
```

Ideally, we pass a valid prop and explain what these props are about. (Some READMEs have a props section for the explanations)

Comment (JustinyAhin) [reply packages/block-editor/src/components/block-patterns-list/README.md]: _Still having to find what the props do exactly_. On my list for next week.

File: packages/block-editor/src/components/block-patterns-list/README.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+		blockPatterns={ shownBlockPatterns }
+		shownPatterns={ shownBlockPatterns }
+		onClickPattern={ onSelectBlockPattern }
+	/>
+);
+```
+
+### Props
+
+#### shownBlockPatterns
```
Comment (youknowriad) [inline packages/block-editor/src/components/block-patterns-list/README.md:34]: this should be blockPatterns I believe.

File: packages/block-editor/src/components/block-patterns-list/README.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+The different [`BlockPattern`](https://github.com/WordPress/gutenberg/blob/cafa3fe8f068c2fd5f3db734ba0b8e5a32ecabad/packages/block-editor/src/components/block-patterns-list/index.js#L15) components that can be shown in the block patterns list.
```
Comment (youknowriad) [inline packages/block-editor/src/components/block-patterns-list/README.md:36]: I don't think this is a component, this is just an array of block pattern objects.

Comment (youknowriad) [reply packages/block-editor/src/components/block-patterns-list/README.md:36]: Same for the "shown" ones. We have two props to support "lazy-rendering".

**Comment by ntsekouras** [comment]
Something has been mixed up here. Can you rebase to include only your changes in the PR?

File: packages/block-editor/src/components/block-patterns-list/README.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+
+-   Type: `Array`
+-   Required: Yes
+
+#### onClickPattern
+
+The performed event after a click on a block pattern. In most cases, the pattern is inserted in the block editor.
+
+-   Type: `any`
```
Comment (gziolo) [inline packages/block-editor/src/components/block-patterns-list/README.md:53]: It's a function:
```suggestion
-   Type: `Function`
```

**Review by gziolo — APPROVED** [review]

**Review by gziolo — APPROVED** [review]
Thank you for the improvements to the block editor documentation :)

**Pull request merged**